server/services/carrier.js

```diff
@@ -483,13 +483,14 @@ module.exports = class CarrierService {
         }
 
         let nextLocation;
-        let distancePerTick;
+        let distancePerTick = this.getCarrierDistancePerTick(game, carrier, warpSpeed, instantSpeed);
+        let distanceToDestination = this.distanceService.getDistanceBetweenLocations(carrier.location, destinationStar.location);
 
-        if (instantSpeed) {
-            distancePerTick = this.distanceService.getDistanceBetweenLocations(carrier.location, destinationStar.location);
+
+        if (instantSpeed || (distancePerTick >= distanceToDestination) ) {
```

Comment:
This is slightly less performant than previous implementation. You now have to calculate `distancePerTick` and `distanceToDestination` **every time**, can you find a way to only do it if `instantSpeed` is false?